### ISSUE TITLE
Add ability to set ValueInputOption in df_to_sheet

### ISSUE THREAD

**Issue by andmatt** [aiguofer/gspread-pandas#26]
Hey Diego,

I recently ran into an issue where I needed to export dataframe containing a `Version Number` column. I noticed that my string value "6.10" was being interpreted as a float and getting converted to "6.1" in the google sheet.

After messing around with `Spread`, I noticed that it was coming from this [line](https://github.com/aiguofer/gspread-pandas/blob/master/gspread_pandas/client.py#L671) from `Spread.update_cells` from `gspread_pandas.client`

```
self._retry_func(partial(self.sheet.update_cells, cells, "USER_ENTERED"))
```

changing the [`value_input_option`](https://developers.google.com/sheets/api/reference/rest/v4/ValueInputOption) parameter from "USER_ENTERED" to "RAW" fixed this issue. 

Would it be possible to change that method to take a kwarg instead?

I was also thinking it might be helpful to add the ability to map the input to certain columns and default the rest to `USER_ENTERED`.

I can also take a stab at it on a personal fork if that'd help - thanks!

**Comment by aiguofer** [comment]
Hey Matt,

I think adding a param to `df_to_sheet` (and `update_cells`) totally makes sense. Gspread uses 'RAW' by default, but I changed it in gspread-pandas to 'USER_ENTERED' in order to get Google Sheet functions to work.

I like the idea of being able to map the input for specific columns, but it would be a bit complicated to implement. The Google sheets API requires a `single value_input_option` per call, even for batch updates. However, I can think of a few ways to accomplish this:

- Make one call to [batch update](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/batchUpdate) per `value_input_option`. I'm not sure how complicated it would be to split up the cells and create these requests since this method isn't implemented in `gspread`. However, this seems like the "most correct" way to do it, especially since it would keep the number of calls to a minimum.

- Make a [regular update](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/update) call per "contiguous" chunk of the same `value_input_type`. 

- Make a regular update call using "USER_ENTERED"  for all values like we do currently, then make another update call with the specified value for each column that needs a different `value_input_option`.

I've been working on some major changes for `v2` and have some local work for that. It'd be awesome to get this into that version. If you have some time to work on it that'd be great, and if not I'll add at least the option to specify the `value_input_option` to `df_to_sheet`.

**Comment by andmatt** [comment]
I have some time to work on this. I can start by trying to implement that first option. When are you aiming to push the release, and which branch should I fork? Thanks!

**Comment by aiguofer** [comment]
That'd be awesome! I'm aiming to push it sometime this week. Once I finish writing tests for the `Client` and `Spread` classes and improve documentation (especially around configuration, credentials, etc) it should be good to go.

I think it'd be easiest if you work off the [v2 branch](https://github.com/aiguofer/gspread-pandas/tree/v2). Note that the `user` param is now optional in that branch and will use "default' by default. I've made a few force pushes to that branch, but I believe anything else I add will be as a new commit on top.

**Comment by aiguofer** [comment]
Closed by #28
